rekallpy/rekall/interval_set.py

```diff
@@ -809,32 +810,63 @@ def coalesce(self,
         Returns:
             A new IntervalSet of intervals that are disjoint along ``axis`` and
             are at least ``epsilon`` apart.
+            
         """
-
-        def update_output(output, intrvl):
-            if len(output) == 0:
-                output.append(intrvl)
-            else:
-                merge_candidate = output[-1]
+        if (len(self._intrvls) == 0):
+            return self
+
+        new_coalesced_intrvls = []
+        current_intrvls = []
+
+        for intrvl in self._intrvls:
+            for cur in current_intrvls:
```

Comment:
I think you can combine this loop at the loop comprehension below to make the code more readable. In this loop you're adding intervals to `new_coalesced_intrvls` if they don't overlap with `intrvl`. In the loop comprehension below you're removing those exact same intervals from `current_intrvls` from `current_intrvls`.

This suggests a format like:
```
new_current_intrvls = []
for cur in current_intrvls:
    if overlaps: new_current_intrvls.append(cur)
    else: new_coalesced_intrvls.append(cur)
```